fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java

```diff
@@ -84,6 +87,54 @@ public static ConnectContext buildConnectContext() {
         return context;
     }
 
+    public static void triggerCollectionOnFirstLoad(Database db, Table table, boolean sync, CountedListener listener) {
+        if (FeConstants.runningUnitTest) {
+            listener.run();
+            return;
+        }
+        if (GlobalStateMgr.getCurrentState().getAnalyzeManager().hasBasicStatsMeta(table.getId())) {
+            listener.run();
+            return;
+        }
+        if (statisticDatabaseBlackListCheck(db.getFullName())) {
+            listener.run();
+            return;
+        }
+        AnalyzeStatus analyzeStatus = new NativeAnalyzeStatus(GlobalStateMgr.getCurrentState().getNextId(),
+                db.getId(), table.getId(), null, StatsConstants.AnalyzeType.FULL,
+                StatsConstants.ScheduleType.ONCE, null, LocalDateTime.now());
+        analyzeStatus.setStatus(StatsConstants.ScheduleStatus.PENDING);
+        GlobalStateMgr.getCurrentAnalyzeMgr().addAnalyzeStatus(analyzeStatus);
+
+        Future<?> future;
+        try {
+            future = GlobalStateMgr.getCurrentAnalyzeMgr().getAnalyzeTaskThreadPool()
+                    .submit(() -> {
+                        StatisticExecutor statisticExecutor = new StatisticExecutor();
+                        ConnectContext statsConnectCtx = StatisticUtils.buildConnectContext();
+                        statsConnectCtx.setThreadLocalInfo();
+
+                        statisticExecutor.collectStatistics(statsConnectCtx,
+                                StatisticsCollectJobFactory.buildStatisticsCollectJob(db, table, null, null,
+                                        StatsConstants.AnalyzeType.FULL, StatsConstants.ScheduleType.ONCE, null),
+                                analyzeStatus,
+                                false);
+                        listener.run();
```

Comment:
listener shuold in `finally block {}` ?

Reply:
if statistics task execute too long, will block Txn status?

Reply:
Yes, it's stupid. Updated.

Reply:
It won't block transaction processing, but it will hold the load progress at 99%.